automatix/command.py

```diff
@@ -103,17 +104,27 @@ def execute(self, interactive: bool = False, force: bool = False):
             if force:
                 return
 
-            err_options = 'p: proceed (default), r: retry, a: abort'
-            if self.env.batch_mode:
-                err_options += ', c: abort & continue to next (CSV processing)'
-
-            err_answer = input(f'[CF] What should I do? ({err_options})\a')
+            err_answer = self._ask_user(question='[CF] What should I do?', allowed_options=['p', 'r', 'a'])
+            # answers 'a' and 'c' are handled by _ask_user, 'p' means just pass
             if err_answer == 'r':
                 return self.execute(interactive)
-            if err_answer == 'a':
-                raise AbortException(return_code)
-            if self.env.batch_mode and err_answer == 'c':
-                raise SkipBatchItemException()
+
+    def _ask_user(self, question: str, allowed_options: list):
+        if self.env.batch_mode:
+            allowed_options.append('c')
+
+        options = ', '.join([f'{k}: {POSSIBLE_ANSWERS[k]}' for k in allowed_options])
+        answer = input(f'{question} ({options})')
+        if answer == '':  # default
+            answer = 'p'
+        if answer not in allowed_options:
```

Comment:
I'd use this as a condition for a while loop instead of recursively calling, but either way works.